docs/sources/flow/reference/components/prometheus.exporter.postgres.md

```diff
@@ -0,0 +1,125 @@
+---
+# NOTE(rfratto): the title below has zero-width spaces injected into it to
+# prevent it from overflowing the sidebar on the rendered site. Be careful when
+# modifying this section to retain the spaces.
+#
+# Ideally, in the future, we can fix the overflow issue with css rather than
+# injecting special characters.
+
+title: prometheus.exporter.postgres
+labels:
+  stage: beta
+---
+
+# prometheus.exporter.postgres
+The `prometheus.exporter.postgres` component embeds
+[postgres_exporter](https://github.com/prometheus-community/postgres_exporter) for collecting metrics from a postgres database.
+
+Multiple `prometheus.exporter.postgres` components can be specified by giving them different 
+labels.
+
+## Usage
+
+```river
+prometheus.exporter.postgres "LABEL" {
+    data_source_names = DATA_SOURCE_NAMES_LIST
+}
+```
+
+## Arguments
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`data_source_names`                  | `list(secret)`      | Specifies the Postgres server(s) to connect to.  |         | yes
+`disable_settings_metrics`           | `bool`              | Disables collection of metrics from pg_settings. | `false` | no
+`disable_default_metrics`            | `bool`              | When `true`, only exposes metrics supplied from `custom_queries_path`. | `false` | no
+`autodiscover_databases`             | `bool`              | Whether to automatically discover databases to scrape metrics from. | `false` | no
+`exclude_databases`                  | `list(string)`      | A list of databases to ignore when `autodiscover_databases` is `true` | | no
+`include_databases`                  | `list(string)`      | Includes only specific databases (excluding all others) when autodiscover_databases is `true` | | no
+`custom_queries_path`                | `string`            | Path to YAML file containing custom queries to expose as metrics. | | no
+
+The format for connection strings in `data_source_names` can be found in the [official postgresql documentation](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING).
+
+See examples for the `custom_queries_path` file in the [postgres_exporter repository](https://github.com/prometheus-community/postgres_exporter/blob/master/queries.yaml).
+
+**NOTE**: There are a number of environment variables that are not recommended for use, as they will affect _all_ `prometheus.exporter.postgres` components. A full list can be found in the [postgres_exporter repository](https://github.com/prometheus-community/postgres_exporter#environment-variables).
```

Comment:
Personally, I would be hesitant to unset any environment variables as that could be pretty unexpected (picturing some shenanigans of having env vars in the config and then triggering a reload, at which point it would error as they had been unset the first go around).

As for outright forbidding it, I lean towards allowing the user to use them if they want while also noting we discourage it and why. It's not a strongly held opinion of mine, though, so would be open to discussion.